packages/wells/src/client/api/surveys.ts

```diff
@@ -0,0 +1,62 @@
+import { Sequence, SequencesAPI } from '@cognite/sdk';
+import { SequenceRow } from 'stable/dist/src/api/sequences/sequenceRow';
+import { Survey, SearchSurveys } from '../model/Survey';
+
+export class Surveys extends SequencesAPI {
+  /**
+   * Maps from sequence array to a survey array.
+   * Contains lazy getter for rows
+   *
+   * @param sequences
+   */
+  private mapToSurvey = (sequences: Sequence[]): Survey[] => {
+    return sequences.map(sequence => {
+      return <Survey>{
+        id: sequence.id,
+        name: sequence.name,
+        description: sequence.description,
+        assetId: sequence.assetId,
+        externalId: sequence.externalId,
+        metadata: sequence.metadata,
+        rows: async (limit: number = 100): Promise<SequenceRow[]> => {
+          return this.retrieveRows({
+            externalId:
+              sequence.externalId != undefined ? sequence.externalId : '',
+          }).autoPagingToArray({ limit: limit });
+        },
+      };
+    });
+  };
+
+  /**
+   * List all trajectories for a particular assetId (wellbore)
+   * and returns a list of surveys
+   *
+   * @param assetId
+   */
+  listTrajectories = async (
```

Comment:
This looks good! Could we also add a method in the Wellbore class so if the user does wellbore.listSurveys(), we would internally call this method with the wellbore id as a parameter?

I don't know how it's implemented in the JS SDK (I am more familiar with the Python one) but it should be sort of similar to what happens when we call asset.children() in the assets API -  There probably is an internal reference to the CDF client in the asset class (like asset._cdfClient) and then the asset itself has a `children` attribute which is lazy, meaning it is not actually filled until you actually call the `children()` method, then it does asset._cdfClient.list to find other assets that have this asset id as parentId.

Also, from what I understood from Julie's presentation, I believe we can have the trajectory metadata already embedded in the wellbore class, so you could do wellbore.trajectory and get that one trajectory. Then the wellbore.listSurveys() method would list all surveys, regardless of being the real trajectory or planned or anything else

Reply:
Do you want an extra method called `wellbore.listSurveys()` or to rename the current method from `wellbore.listTrajectories()` to `wellbore.listSurveys()`. And should `wellbore.trajectory` list the trajectory that has actually been drilled?

The listTrajectories already calls the for the trajectories internally based on the wellbore id: 
![image](https://user-images.githubusercontent.com/25263842/98252045-cb76da80-1f79-11eb-9975-bbdcbc11c43a.png)

in the case of wellbores, the `asset.id` is the wellbore id and `asset.parentId` is reference to the well Id for that wellbore

Reply:
This is how you would find your wellbores, trajectories and rows based on a wellId: 
![image](https://user-images.githubusercontent.com/25263842/98253877-df234080-1f7b-11eb-9773-825816436a19.png)